commit 0b770193b214d527982ee5dcf73b4538d3bc8cb5
Author: Vincent Bernat <bernat@luffy.cx>
Date:   Wed Sep 1 15:36:21 2010 +0200

    Do not set reload flag in the main process.
    
    The main process does not use the reload flag. When setting the reload
    flag in the main process and a child dies, the new child will have the
    reload flag for a short time. There seems to be some race condition
    here. This "fix" seems to circumvent this race condition.

diff --git a/keepalived/core/main.c b/keepalived/core/main.c
index ef472f8..5b6ec7a 100644
--- a/keepalived/core/main.c
+++ b/keepalived/core/main.c
@@ -87,9 +87,6 @@ start_keepalived(void)
 void
 sighup(void *v, int sig)
 {
-	/* Set the reloading flag */
-	SET_RELOAD;
-
 	/* Signal child process */
 	if (vrrp_child > 0)
 		kill(vrrp_child, SIGHUP);
